app/src/androidTest/java/com/duckduckgo/app/httpsupgrade/HttpsUpgraderTest.kt

```diff
@@ -68,6 +70,13 @@ class HttpsUpgraderTest {
         assertFalse(testee.shouldUpgrade(Uri.parse("http://www.local.url")))
     }
 
+    @Test
+    fun whenHttpDomainIsUserWhitelistedThenShouldNotUpgrade() {
+        whenever(mockUserWhitelistDao.contains("www.local.url")).thenReturn(true)
+        bloomFilter.add("www.local.url")
```

Comment:
No, we need to add the url to the bloom filter so it is in our upgrade list and then test that it's not upgraded despite this because of the user whitelist. The test should fail before the new user whitelist logic is added (red) and then pass after (green). If we remove adding it to the bloom filter then the test passes even before the new logic is added (or if we comment it out) meaning that it does nothing to test the new logic.

I like your pixel test suggestion. Added!

Reply:
aha! I really enjoyed this explanation, you got me thinking about it for a while. I really liked the green/red reference 👍